libs/portals/admin/regulations-admin/src/state/useDraftingState.ts

```diff
@@ -0,0 +1,423 @@
+              ministry: draft.ministry.value,
+              draftingNotes: draft.draftingNotes.value,
+              idealPublishDate: toISODate(draft.idealPublishDate?.value),
+              fastTrack: draft.fastTrack.value,
+              lawChapters: draft.lawChapters.value,
+              signatureDate: toISODate(draft.signatureDate.value),
+              signatureText: draft.signatureText.value,
+              effectiveDate: toISODate(draft.effectiveDate.value),
+              type: draft.type.value,
+              draftingStatus: newStatus || draft.draftingStatus,
+              signedDocumentUrl: draft.signedDocumentUrl.value,
+            },
+          },
+        },
+      })
+        .then((res) => {
+          if (res.errors && res.errors.length > 1) {
+            throw res.errors[0]
+          }
+          return { success: true, error: undefined }
+        })
+        .catch((error) => {
+          return { success: false, error: error as Error }
+        })
+
+    const actions = {
+      goBack: prevStep
+        ? () => {
+            actions.goToStep(prevStep)
+          }
+        : undefined,
+
+      goForward: nextStep
+        ? () => {
+            // BASICS – It's pointless (albeit harmless) to go forward if title and text are empty
+            if (
+              step.name === 'basics' &&
+              (draft.title.error || draft.text.error)
+            ) {
+              // trigger dirty=true on both title and text
+              !draft.title.dirty &&
+                actions.updateState('title', draft.title.value)
+              !draft.text.dirty && actions.updateState('text', draft.text.value)
+              return // Prevent the user going forward
+            }
+
+            const isTitleAmending =
+              findRegulationType(draft.title.value) === 'amending'
+
+            const amendingTitleAndBaseType =
+              isTitleAmending && draft.type.value === 'base'
+
+            const baseTitleAndAmendingType =
+              !isTitleAmending && draft.type.value === 'amending'
+            if (
+              step.name === 'basics' &&
+              (amendingTitleAndBaseType || baseTitleAndAmendingType)
+            ) {
+              return // Prevent the user going forward
+            }
+
+            const isTitleTooLong = draft.title.value.length > 1024
+            if (isTitleTooLong) {
+              return // Prevent the user going forward
+            }
+
+            actions.goToStep(nextStep)
+          }
+        : undefined,
+
+      goToStep: async (stepName: Step) => {
+        if (!isDraftEmpty(draft)) {
+          await actions.saveStatus(true)
+        }
+        navigate(getEditUrl(draft.id, stepName))
+      },
+
+      // FIXME: rename to updateProp??
+      updateState: <Prop extends RegDraftFormSimpleProps>(
+        name: Prop,
+        value: RegDraftForm[Prop]['value'],
+      ) => {
+        // @ts-expect-error  (FML! FIXME: make this nicer)
```

Comment:
Fixme!

Reply:
https://github.com/island-is/island.is/pull/8983#discussion_r1123112458